### PR TITLE
M2-5794: Dataviz - Date Item support

### PR DESCRIPTION
**Pull request by alex-i-e**
- [x] Tests for the changes have been added
- [ ] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

🔗 [Jira Ticket M2-5794](https://mindlogger.atlassian.net/browse/M2-5794)

### 📸 Screenshots

| Before (Optional)                      | After                                 |
| -------------------------------------- | ------------------------------------- |
| image | image |
| image | image |

### CODE REVIEW

File: src/modules/Dashboard/features/RespondentData/RespondentData.utils.ts [hunk 1]
```diff
@@ -11,3 +15,18 @@ export const createArrayForSlider = ({
     value: minValue + index,
     label: minValue + index,
   }));
+
+export const getDateForamttedResponse = (answer: DecryptedDateAnswer) => {
+  if (!answer?.value) return '';
+
+  const day = answer.value.day;
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentData.utils.ts]:
```suggestion
  const { day, month, year } = answer.value;
```

File: src/shared/components/DatePicker/DatePicker.styles.ts [hunk 2]
```diff
@@ -225,3 +225,7 @@ export const DatePickerFallback = styled(Box)`
   min-height: 30rem;
   min-width: 30rem;
 `;
+
+export const StyledSpan = styled('span')`
```
Comment (IhorNesterenko99) [inline src/shared/components/DatePicker/DatePicker.styles.ts:229]: Why do you need the span?

File: src/shared/components/DatePicker/DatePickerContainer/DatePickerContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { useTranslation } from 'react-i18next';
+
+import { Svg } from 'shared/components/Svg';
+
+import { StyledIconBtn, StyledTextField } from '../DatePicker.styles';
+import { DatePickerContainerProps } from './DatePickerContainer.types';
+
+export const DatePickerContainer = ({
```
Comment (IhorNesterenko99) [inline src/shared/components/DatePicker/DatePickerContainer/DatePickerContainer.tsx]: Maybe rename it to DatePickerTextField or something similar?

File: src/shared/components/DatePicker/DatePicker.styles.ts (hunk 2 above)
Comment (alex-i-e) [reply src/shared/components/DatePicker/DatePicker.styles.ts:229]: The tooltip throw the warning in console, it requires an element wrapper. 
**_Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?_
_Check the render method of `ForwardRef(Tooltip)`..._**

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**